pkg/backup/writer/abs_writer.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writer
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/coreos/etcd-operator/pkg/backup/util"
+	"github.com/pborman/uuid"
+)
+
+var _ Writer = &absWriter{}
+
+type absWriter struct {
+	abs *storage.BlobStorageClient
+}
+
+// NewABSWriter creates a abs writer.
+func NewABSWriter(abs *storage.BlobStorageClient) Writer {
+	return &absWriter{abs}
+}
+
+const (
+	// AzureBlobBlockChunkLimitInBytes 100MiB is the limit
+	AzureBlobBlockChunkLimitInBytes = 104857600
+)
+
+// Write writes the backup file to the given abs path, "<abs-container-name>/<key>".
+func (absw *absWriter) Write(path string, r io.Reader) (int64, error) {
+	container, key, err := util.ParseBucketAndKey(path)
+	if err != nil {
+		return 0, err
+	}
+
+	containerRef := absw.abs.GetContainerReference(container)
+	containerExists, err := containerRef.Exists()
+	if err != nil {
+		return 0, err
+	}
+
+	if !containerExists {
+		return 0, fmt.Errorf("container %v does not exist", container)
+	}
+
+	blob := containerRef.GetBlobReference(key)
+	err = blob.CreateBlockBlob(&storage.PutBlobOptions{})
+	if err != nil {
+		return 0, err
+	}
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(r)
```

Comment:
what if reader has a 2 GB? will `ReadFrom` return `ErrTooLarge` on system that has smaller memory or disk? 
I think we need to figure out a way to read 100MB into buffer and then write 100MB into ABS.

Reply:
@rjtsdl this is just an optimization and it will not affect the correctness. we can optimize the uploading logic in a future pr.

Reply:
I will submit this optimization in the next pr since I dont want to block the merging of this current pr

Reply:
Thx, we can do it in another PR. :)